src/Stratis.Bitcoin.Features.MemoryPool/MempoolOrphans.cs

```diff
@@ -347,11 +360,15 @@ public class OrphanTx
         /// <returns>Whether the orphan transaction was added.</returns>
         public Task<bool> AddOrphanTx(ulong nodeId, Transaction tx)
         {
+            this.mempoolLogger.LogTrace("{0}:{1}, {2}:'{3}'", nameof(nodeId), nodeId, nameof(tx), tx?.GetHash());
             return this.MempoolLock.WriteAsync(() =>
```

Comment:
Please assign the result to bool, add exit log with this bool

Reply:
This is returns a task so exit may have not occurred yet.

Reply:
Yes that is why you create a variable for the result and you will assign the task to it, only then you create exit log and only then you return from the method.